encoding/json/decode.go

```diff
@@ -908,13 +909,23 @@ func (d *Decoder) decodeFieldType(valueJSON any, results typeDecodingResults) ca
 	)
 }
 
-func (d *Decoder) decodeFunctionType(returnValue, parametersValue, id any, results typeDecodingResults) cadence.Type {
+func (d *Decoder) decodePurity(purity any) cadence.FunctionPurity {
+	functionPurity := toString(purity)
```

Comment:
Are there any tests for the field non-existing case? `toString()` seems to be panicking if the value is not a string..